### ISSUE TITLE
2019-12-06 M1SD - identify parameters that if missing from PARAMETERLIST cause crashes

### ISSUE THREAD

**Issue by tensfeldt** [tensfeldt/openNCA#87]
## CMAX alone specified
This should produce a parameter result with simply CMAX but fails complaining that kel_v isn't found. The dependency for kel_v is not appropriate for this.

```
> mct$PARAMETERLIST <- "CMAX"
> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
Error in value[[3L]](cond) : 
  Error in grep(x, parameter_list, ignore.case = TRUE, perl = TRUE): object 'kel_v' not found
For SDEID 70745987
In addition: Warning messages:
1: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELRSQ'
2: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELNOPT'
3: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTP'
4: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTPREDOSECRIT' cannot be computed if 'CMAXi' is not part of the calculated parameters
5: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in grep(x, parameter_list, ignore.case = TRUE, perl = TRUE): object 'kel_v' not found
For SDEID 70745987 
```

##  CMAX/TMAX/AUCLAST  only specified 
fails with kel_v not found
```
> mct$PARAMETERLIST <- "CMAX;TMAX;AUCLAST"
> results_list <- run_computation(data=d, map=mct, flag=flags, parameterset=parameterset)
Error in value[[3L]](cond) : 
  Error in grep(x, parameter_list, ignore.case = TRUE, perl = TRUE): object 'kel_v' not found
For SDEID 70745987
In addition: Warning messages:
1: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELRSQ'
2: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELNOPT'
3: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTP'
4: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTPREDOSECRIT' cannot be computed if 'CMAXi' is not part of the calculated parameters
5: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in grep(x, parameter_list, ignore.case = TRUE, perl = TRUE): object 'kel_v' not found
For SDEID 70745987 

```
## set PARAMETERLIST to a full list and then remove one parameter, in order at a time. 
What is shown below is the results of each failure. If removal of a parameter caused a crash, the parameter was added to the end of the list and the run_computation was re-run, in an attempt to identify sensitive parameters, i.e. parameters for which the dependencies are not properly resolved in the code for computation and reporting.

from 2019-12-06_M1SD.json MCT

```
mct$PARAMETERLIST <- paste(c("AUCALL","AUCALLDN","AUCINFO","AUCINFOC","AUCINFODN","AUCINFP","AUCINFPC","AUCINFPDN","AUCLAST","AUCLASTDN","AUCLASTC","AUCT","AUCT1_T2","AUCXPCTO","AUCXPCTP","AUMCINFO","AUMCINFP","AUMCLAST","AUMCXPTO","AUMCXPTP","CLAST","CEST","CLFO","CLFOW","CLFP","CLFPW","CMAX","CMAXDN","CMIN","DOSEC","DOSE","KEL","KELNOPT","KELRSQ","KELRSQA","KELTMHI","KELTMLO","LASTTIME","MRTLAST","MRTEVIFO","MRTEVIFP","THALF","THALFF","TLAG","TLAST","TMAX","TMIN","VZFO","VZFOW","VZFP","VZFPW","FLGACCEPTKEL","FLGACCEPTKELCRIT","FLGACCEPTPREDOSE","FLGACCEPTPREDOSECRIT","FLGACCEPTTMAX","SPANRATIOCRIT"), collapse=";")

*** i:  CLAST  removed
[1] "CEST;CLFO;CLFOW;CLFP;CLFPW;CMAX;CMAXDN;CMIN;DOSEC;DOSE;KEL;KELNOPT;KELRSQ;KELRSQA;KELTMHI;KELTMLO;LASTTIME;MRTLAST;MRTEVIFO;MRTEVIFP;THALF;THALFF;TLAG;TLAST;TMAX;TMIN;VZFO;VZFOW;VZFP;VZFPW;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT"
Error in value[[3L]](cond) : 
  Error in auc_inf_o(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'c_last' not found
For SDEID 70745987
In addition: There were 50 or more warnings (use warnings() to see the first 50)


*** i:  MRTLAST  removed
[1] "MRTEVIFO;MRTEVIFP;THALF;THALFF;TLAG;TLAST;TMAX;TMIN;VZFO;VZFOW;VZFP;VZFPW;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT;CLAST"
Error in value[[3L]](cond) : 
  Error in auc_inf_o(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'auclast' not found
For SDEID 70745987
In addition: There were 50 or more warnings (use warnings() to see the first 50)

*** i:  TLAST  removed
[1] "TMAX;TMIN;VZFO;VZFOW;VZFP;VZFPW;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT;CLAST;MRTLAST"
Error in value[[3L]](cond) : 
  Error in auc_last(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_last' not found
For SDEID 70745987
In addition: There were 50 or more warnings (use warnings() to see the first 50)

*** i:  VZFOW  removed
[1] "VZFP;VZFPW;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT;CLAST;MRTLAST;TLAST"
Error in value[[3L]](cond) : 
  Error in auc_XpctO(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'aucinf_o' not found
For SDEID 70745987
In addition: There were 50 or more warnings (use warnings() to see the first 50)

*** i:  FLGACCEPTTMAX  removed
[1] "SPANRATIOCRIT;CLAST;MRTLAST;TLAST;VZFOW"
Error in value[[3L]](cond) : 
  Error in auc_last(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_max' not found
For SDEID 70745987
In addition: There were 50 or more warnings (use warnings() to see the first 50)

*** i:  CLAST  removed
[1] "MRTLAST;TLAST;VZFOW;FLGACCEPTTMAX"
Error in value[[3L]](cond) : 
  Error in auc_inf_o(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'c_last' not found
For SDEID 70745987
In addition: There were 20 warnings (use warnings() to see them)

*** i:  MRTLAST  removed
[1] "TLAST;VZFOW;FLGACCEPTTMAX;CLAST"
Error in value[[3L]](cond) : 
  Error in auc_inf_o(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'auclast' not found
For SDEID 70745987
In addition: Warning messages:
1: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELRSQ'
2: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELNOPT'
3: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTP'
4: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGEMESIS' is not present in the dataset
5: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTPREDOSECRIT' cannot be computed if 'CMAXi' is not part of the calculated parameters
6: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in auc_inf_o(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 'auclast' not found
For SDEID 70745987 

*** i:  TLAST  removed
[1] "VZFOW;FLGACCEPTTMAX;CLAST;MRTLAST"
Error in value[[3L]](cond) : 
  Error in auc_last(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_last' not found
For SDEID 70745987
In addition: Warning messages:
1: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELRSQ'
2: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELNOPT'
3: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTP'
4: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGEMESIS' is not present in the dataset
5: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTPREDOSECRIT' cannot be computed if 'CMAXi' is not part of the calculated parameters
6: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in auc_last(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_last' not found
For SDEID 70745987 

*** i:  VZFOW  removed
[1] "FLGACCEPTTMAX;CLAST;MRTLAST;TLAST"
Error in value[[3L]](cond) : 
  Error in grep(x, parameter_list, ignore.case = TRUE, perl = TRUE): object 'kel_v' not found
For SDEID 70745987
In addition: Warning messages:
1: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELRSQ'
2: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELNOPT'
3: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTP'
4: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGEMESIS' is not present in the dataset
5: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTPREDOSECRIT' cannot be computed if 'CMAXi' is not part of the calculated parameters
6: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in grep(x, parameter_list, ignore.case = TRUE, perl = TRUE): object 'kel_v' not found
For SDEID 70745987 

*** i:  FLGACCEPTTMAX  removed
[1] "CLAST;MRTLAST;TLAST;VZFOW"
Error in value[[3L]](cond) : 
  Error in auc_last(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_max' not found
For SDEID 70745987
In addition: Warning messages:
1: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELRSQ'
2: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'KELNOPT'
3: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTKELCRIT' values provided via 'map' does not have a valid parameter name 'AUCXPCTP'
4: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
  Flag 'FLGACCEPTPREDOSECRIT' cannot be computed if 'CMAXi' is not part of the calculated parameters
5: In run_M1_SS_computation(data = merged_data, map = map_data, method = method,  :
 
 Show Traceback
 
 Rerun with Debug
 Error in value[[3L]](cond) : 
  Error in auc_last(conc = tmp_df[, map_data$CONC], time = tmp_df[, map_data$TIME], : object 't_max' not found
For SDEID 70745987 

```

**Comment by rudraya** [comment]
`This issue was addressed in fce7a3c, please let me know if it still occurs.

**Comment by jhhughes256** [comment]
### Update

Testing results for above examples as of commit fce7a3c. Tick represents success (i.e. error can no longer be replicated).

- [ ] CMAX
- [ ] CMAX;TMAX;AUCLAST
- [ ] "CEST;CLFO;CLFOW;CLFP;CLFPW;CMAX;CMAXDN;CMIN;DOSEC;DOSE;KEL;KELNOPT;KELRSQ;KELRSQA;KELTMHI;KELTMLO;LASTTIME;MRTLAST;MRTEVIFO;MRTEVIFP;THALF;THALFF;TLAG;TLAST;TMAX;TMIN;VZFO;VZFOW;VZFP;VZFPW;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT" (CLAST removed)
- [ ] "MRTEVIFO;MRTEVIFP;THALF;THALFF;TLAG;TLAST;TMAX;TMIN;VZFO;VZFOW;VZFP;VZFPW;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT;CLAST" (MRTLAST removed)
- [ ] "TMAX;TMIN;VZFO;VZFOW;VZFP;VZFPW;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT;CLAST;MRTLAST" (TLAST removed)
- [ ] "VZFP;VZFPW;FLGACCEPTKEL;FLGACCEPTKELCRIT;FLGACCEPTPREDOSE;FLGACCEPTPREDOSECRIT;FLGACCEPTTMAX;SPANRATIOCRIT;CLAST;MRTLAST;TLAST" (VZFOW removed)
- [ ] "SPANRATIOCRIT;CLAST;MRTLAST;TLAST;VZFOW" (FLGACCEPTTMAX removed)
- [ ] "MRTLAST;TLAST;VZFOW;FLGACCEPTTMAX" (CLAST removed)
- [ ] "TLAST;VZFOW;FLGACCEPTTMAX;CLAST" (MRTLAST removed)
- [ ] "VZFOW;FLGACCEPTTMAX;CLAST;MRTLAST" (TLAST removed)
- [ ] "FLGACCEPTTMAX;CLAST;MRTLAST;TLAST" (VZFOW removed)
- [ ] "CLAST;MRTLAST;TLAST;VZFOW" (FLGACCEPTTMAX removed)
